src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -67,6 +68,14 @@ def set_notifications
     @notifications = filter_notifications_by_kind(@notifications, @filter_kind)
   end
 
+  def set_notifications_count
+    @counted_notifications = {}
+    @counted_notifications['all'] = @notifications.count
```

Comment:
`@notifications` is already filtered by `set_notifications` at this point. So we count only the subset.

Reply:
Ah, I see, that's a typo. It was supposed to be `User.notifications.count`. Then I wonder if that would trigger another query. Maybe I'll move the ` @counted_notifications['all'] =` right after `@notifications = User.session!.notifications.for_web.includes(notifiable: [{ commentable: [{ comments: :user }, :project, :bs_request_actions] }, :bs_request_actions, :reviews])` before the filtering part

Reply:
https://github.com/openSUSE/open-build-service/pull/16257/commits/41ce5d20227116acd47263a62bf3354cbd2ad22b